lib/adapters/local/find/query-planner.js

```diff
@@ -131,6 +131,10 @@ function checkFieldsLogicallySound(indexFields, selector) {
   var firstField = indexFields[0];
   var matcher = selector[firstField];
 
+  if (matcher === undefined) {
```

Comment:
`typeof matcher === 'undefined'` is safer because some JS engines allow overwriting `undefined` (i.e. it's a variable not a keyword)